packages/sdk/src/contracts/prebuilt-implementations/marketplace.ts

```diff
@@ -296,6 +298,44 @@ export class Marketplace implements UpdateableNetwork {
     return this.contractWrapper.readContract.timeBuffer();
   }
 
+  /**
+   * Get all the offers
+   *
+   * @remarks Fetch all the offers for a specified direct listing.
+   * @example
+   * ```javascript
+   * const offers = await marketplaceContract.getOffers(listingId);
+   * const firstOffer = offers[0];
+   * ```
+   *
+   * @param listingId - the id of the listing to fetch offers for
+   */
+  public async getOffers(listingId: BigNumberish): Promise<Offer[]> {
+    // get all new offer events from this contract
+    const events = await this.events.getEvents<NewOfferEventObject>("NewOffer");
+    // get only the events for this listing id
+    const listingEvents = events.filter((e) => e.data.listingId.eq(listingId));
+    // derive the offers from the events
+    const offers = await Promise.all(
+      listingEvents.map(async (e): Promise<Offer> => {
+        const offer = await mapOffer(
+          this.contractWrapper.getProvider(),
+          BigNumber.from(listingId),
+          {
+            quantityWanted: e.data.quantityWanted,
+            pricePerToken: e.data.totalOfferAmount.div(e.data.quantityWanted),
+            currencyContractAddres: e.data.currency,
+            buyerAddress: e.data.offeror,
+            currency: e.data.currency,
+            listingId: e.data.listingId,
+          },
```

Comment:
oof i didn't realize this offer parameter can be any, we should really type that. It's letting a few type errors go unnoticed:

the passed in `offer` object should have:

```ts
{
    quantityDesired: BigNumberish,
    pricePerToken: BigNumberish,
    currency: string,
    offeror: string,
    quantityWanted: BigNumberish
}
```
so no `buyerAddress` and no `currencyContractAddres`

feel free to add a type for this 🙏

Reply:
@joaquim-verges I thought the purpose of the `mapOffer` function was to take input of "any" and return that payload as type "Offer". If that's the case, wouldn't forcing the input to be of type "Offer" defeat the purpose?

Reply:
The goal of mapOffer was to transform a contract struct to a more dev friendly Offer type. Now you're adding a way to map from a contract event to a Offer type.

So the inputs to mapOffer are not quite the output type

Reply:
Forgive my n00bishness with TypeScript. 

Is what I did here in commit [1933294](https://github.com/thirdweb-dev/js/pull/191/commits/19332946d6b8f3146e62d03e101d031b01099e4d) what you were looking for?

I suspect it isn't because I had to add some optional keys for the newly added type to support backwards compatibility.

I was thinking about creating a new `mapOffer` function specifically for my use case and I'm happy to do that if it is a preferred path forward.